### DISCUSSION TITLE
Form field attributes?

### DISCUSSION THREAD

**Discussion by nemo-omen** [emmett-framework/emmett#475] [Q&A]
First, I want to say that I'm going through the Emmett docs and I just love this framework and your approach. Thank you for making such a fantastic piece of software!

Now, I have a question. Is there a way to define attributes on `Form` fields so they carry over to the HTML form's inputs? If not, I can just write the markup and take care of things in the template.

**Comment by gi0baro** [comment]
@nemo-omen thank you very much!

Even if you probably can set an attribute to fields with setattr/setitem, I wouldn't go that path as there's quite a lot running under the hood with fields. Since you're main point is in regards HTML, I would probably suggest to go with custom widgets and/or `FormStyle` – which is not documented so much (yet?) unfortunately.

Maybe if you can explain with more details your need I can support you with additional advices :)

**Reply by nemo-omen** [reply]
I'm experimenting with [HTMX](https://htmx.org), which uses attributes on HTML tags to interact with server routes. In this case, it does seem more practical to do what I need in the template markup since it looks like a `Form` is much more than a regular HTML form.
